caffe2/core/tensor.h

```diff
@@ -188,7 +188,8 @@ class CAFFE2_API Tensor final {
   // swap method swaps the CONTENTS of the tensors, while std::swap
   // swaps the POINTERS.
   void swap(const Tensor& other) const noexcept {
-    impl_.get()->swap(*other.impl_.get());
+    // NB: use get() to get a non-const pointer!
+    std::swap(*impl_.get(), *other.impl_.get());
```

Comment:
Tbh, I don't like that behavior. It's not changed by that diff, but I'd expect a.swap(b) to do the same as std::swap(a, b). Can we maybe rename this to swapContent()?

Also, where is swap(TensorImpl, TensorImpl) defined? Didn't find it.

Reply:
It's dumb but if we want to fix it, we need to fix all the use sites, and swap is not an easy identifier to codemod for. I put this on future work.

I purposely didn't define a swap(TensorImpl, TensorImpl), because you should just use `std::swap` in that case.

Reply:
Don't we have a clang-based codemod that knows variable types and only renames the swap member of that one type?

Reply:
I know there's an AST based clang codemodder. I don't know about a type directed codemod. I'd love to know about one if there was one.